File: C++/Leetcode 11: Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int max_area = 0;

        while(left < right){
            int area = (right - left) * min(height[left], height[right]);
            if(area > max_area){
                max_area = area;
            }
            if(height[left] > height[right]){
                right -= 1;
            }else{
                left += 1;
            }
        }
        return max_area;
    }
};

File: python/Leetcode 11: Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        area = 0

        while left < right:
            # Calculate current area
            current_area = (right - left) * min(height[left], height[right])
            area = max(area, current_area)

            # Move the pointer with the smaller height
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return area
